contracts/rewards/src/state.rs

```diff
@@ -0,0 +1,389 @@
+use std::collections::HashMap;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Storage, Uint256};
+use cw_storage_plus::{Item, Map};
+use error_stack::{Result, ResultExt};
+
+use crate::{error::ContractError, msg::RewardsParams};
+
+#[cw_serde]
+pub struct StoredParams {
+    pub params: RewardsParams,
+    /// epoch in which the params were updated
+    pub updated: Epoch,
+}
+
+#[cw_serde]
+pub struct EpochTally {
+    pub epoch_num: u64,
+    pub contract: Addr,
+    pub event_count: u64,
+    pub participation: HashMap<Addr, u64>,
+    pub rewards_rate: Uint256,
```

Comment:
```suggestion
    pub distributed_rewards: Uint256,
```
and move it to `Epoch`